mitodl/open-discussions
# Moira integration

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #2621
Closes #2634

#### What's this PR do?
- Adds a MoiraList model
- Adds functions to retrieve a user's moira lists and a moira list's users
- Modifies ChannelMembershipConfig to support moira lists
- Adds functions/tasks to update channel memberships based on moira lists

#### How should this be manually tested?
- Set the following in your `.env` file:
  ```
  FEATURE_MOIRA=True
  MIT_WS_CERTIFICATE=<same as OVS-CI>
  MIT_WS_PRIVATE_KEY=<same as OVS-CI>
  ```
- Create a new channel in the django shell:
  ```
  from channels.api import get_admin_api
  api = get_admin_api()
  api.create_channel("ml1", title="managed moira channel 1", channel_type="private", 
  "membership_is_managed=True)
  ```

- In django admin, create a new ChannelMembershipConfig with query `{"moira_lists": ["<name_of_a_real_moira_list>"]}` and apply it to the new channel above.  An example real list is 'odl-engineering' which you should be a member of.

<img width="441" alt="Screen Shot 2020-03-17 at 3 27 33 PM" src="https://user-images.githubusercontent.com/187676/76894212-16fe3c00-6864-11ea-8167-5b10f4ee6888.png">


- Log in as a user who belongs on the moira list used above.  A celery task should be kicked off, and after completion, if you reload the page, your new channel should show up in the left side bar.
